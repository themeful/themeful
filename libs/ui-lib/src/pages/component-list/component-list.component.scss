:host {
  width: 100%;
  height: 100%;
}

.component-list__wrapper {
  padding: 16px;
}
