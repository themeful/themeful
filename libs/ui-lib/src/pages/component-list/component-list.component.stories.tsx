import { h } from '@stencil/core'
import { action } from '@storybook/addon-actions'
import { of } from 'rxjs'
import sample from './component-list.sample.json'

export default {
  title: 'Pages/Component List',
  args: {
    componentListBundle$: of(sample),
    onAction: action('triggered action'),
  },
}

export const componentList = (args) => {
  return (
    <div>
      <div class="header" style={{ marginBottom: '16px' }}>
        <h1>Component List</h1>
      </div>
      <div class="content">
        <div style={{ position: 'relative' }}>
          <tf-component-list {...args} />
        </div>
      </div>
    </div>
  )
}
