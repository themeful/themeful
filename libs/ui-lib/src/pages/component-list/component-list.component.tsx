import { Component, Event, EventEmitter, h, Host, Prop } from '@stencil/core'
import { ComponentListBundle, FormIntegrationActions } from '@typings'
import { Observable, Subject, Subscription } from 'rxjs'
import '../../components/button'
import '../../components/icon'
import '../../components/menu'
import '../../components/navigation'
import '../../components/property'
import '../../forms/form-integration'

@Component({
  tag: 'tf-component-list',
  styleUrl: 'components.component.scss',
  shadow: true,
})
export class ComponentListComponent {
  /** Style Guides */
  @Prop() themeBundle$: Observable<ComponentListBundle>

  private formData$ = new Subject()
  private nav = []

  private sub = new Subscription()

  /** Event emitted when an action is triggered */
  @Event({ composed: false }) action: EventEmitter<FormIntegrationActions>

  private onAction = ({ detail }): void => {
    if (detail.action !== 'close') {
      this.action.emit(detail)
    }
  }

  public disconnectedCallback(): void {
    this.sub.unsubscribe()
  }

  public render(): HTMLTfThemesElement {
    return (
      <Host>
        <stencil-route-title pageTitle="Themeful - Components" />
        <tf-navigation items={this.nav} size="small" />
        List of Components
        <tf-form-integration {...{ formData$: this.formData$, onAction: this.onAction }} />
      </Host>
    )
  }
}
