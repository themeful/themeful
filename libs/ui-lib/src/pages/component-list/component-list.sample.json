[
  {
    "C3rHXN": {
      "name": "Button",
      "files": [
        {
          "path": "sample/components/lib1/button/button.component.scss",
          "ignored": false
        }
      ],
      "ignored": false
    },
    "iX5ZPn": {
      "name": "Teaser Card",
      "files": [
        {
          "path": "sample/components/lib1/teaser-card/teaser-card.component.scss",
          "ignored": false
        }
      ],
      "ignored": false
    },
    "zBIt36": {
      "name": "Card",
      "files": [
        {
          "path": "sample/components/lib2/card/card.component.scss",
          "ignored": false
        }
      ],
      "ignored": false
    }
  },
  {
    "atBaseFontColor": {
      "component": null,
      "files": null,
      "properties": ["color"],
      "extern": true,
      "crawled": false
    },
    "atButtonBackground": {
      "component": ["Button"],
      "files": ["libs/components/src/lib/button/button.component.scss"],
      "properties": ["background-color"],
      "extern": true,
      "crawled": true
    },
    "atButtonBackgroundHover": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["background-color"],
      "extern": false,
      "crawled": true
    },
    "atButtonBorderRadius": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["border-radius"],
      "extern": false,
      "crawled": true
    },
    "atButtonFontColor": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["color"],
      "extern": false,
      "crawled": true
    },
    "atButtonFontSize": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["font-size"],
      "extern": false,
      "crawled": true
    },
    "atButtonHorizontalSpace": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["padding"],
      "extern": false,
      "crawled": true
    },
    "atButtonVerticalSpace": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["padding"],
      "extern": false,
      "crawled": true
    },
    "atCardBackground": {
      "component": ["Card"],
      "files": ["sample/components/card/card.component.scss"],
      "properties": ["background-color"],
      "extern": false,
      "crawled": true
    }
  },
  {
    "styleGuide1_dark": {
      "name": "Dark",
      "styleGuide": "styleGuide1",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide1_action_tertiary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide1_action_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_tiny"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_light"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide1_content_primary"
          },
          "styleGuide1_mediaQuery_aboveSmallDesktop": {
            "direct": {
              "value": "13px",
              "type": "font-size"
            }
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide1_font_primary"
          }
        }
      }
    },
    "styleGuide1_light": {
      "name": "Light",
      "styleGuide": "styleGuide1",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide1_action_primary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide1_action_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_small"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_black"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide1_content_primary"
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide1_font_primary"
          }
        }
      }
    },
    "styleGuide2_dark": {
      "name": "Dark",
      "styleGuide": "styleGuide2",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide2_brand_tertiary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide2_brand_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_small"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_light"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide2_content_normal"
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide2_font_primary"
          }
        }
      }
    },
    "styleGuide2_light": {
      "name": "Light",
      "styleGuide": "styleGuide2",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide2_brand_primary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide2_brand_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_tiny"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_black"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide2_content_normal"
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide2_font_primary"
          }
        }
      }
    },
    "styleGuide3_default": {
      "name": "Default",
      "styleGuide": "styleGuide3",
      "styles": {
        "dtActionBg": {
          "default": {
            "direct": {
              "value": "#aaaaaa",
              "type": "color"
            }
          },
          "global_mediaQuery_aboveSmallMobile": {
            "direct": {
              "value": "rgba(180, 180, 180, 0.8)",
              "type": "color"
            }
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide3_base_primary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_tiny"
          },
          "global_mediaQuery_aboveSmallMobile": {
            "style": "global_space_small"
          },
          "global_mediaQuery_aboveMobile": {
            "style": "global_space_medium"
          },
          "global_mediaQuery_aboveTablet": {
            "style": "global_space_large"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "styleGuide3_base_primary"
          }
        },
        "dtFontSize100": {
          "global_mediaQuery_aboveSmallMobile": {
            "direct": {
              "type": "font-size",
              "value": "1rem"
            }
          },
          "global_mediaQuery_aboveMobile": {
            "direct": {
              "type": "font-size",
              "value": "2rem"
            }
          }
        }
      }
    }
  }
]
