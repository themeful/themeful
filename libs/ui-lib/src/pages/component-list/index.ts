export * from './component-list.component'
