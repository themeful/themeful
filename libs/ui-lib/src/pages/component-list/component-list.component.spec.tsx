import { newSpecPage } from '@stencil/core/testing'
import { ComponentListComponent } from './component-list.component'

describe('ComponentListComponent', () => {
  it('renders', async () => {
    const { root } = await newSpecPage({
      components: [ComponentListComponent],
      html: '<tf-component-list></tf-component-list>',
    })
    expect(root).toMatchSnapshot()
  })

  it('renders with values', async () => {
    const { root } = await newSpecPage({
      components: [ComponentListComponent],
      html: `<tf-component-list></tf-component-list>`,
    })
    expect(root).toMatchSnapshot()
  })
})
