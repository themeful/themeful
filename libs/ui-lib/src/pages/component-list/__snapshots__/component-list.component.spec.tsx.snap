// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentListComponent renders 1`] = `
<tf-component-list>
  <mock:shadow-root>
    <stencil-route-title pagetitle="Themeful - Components"></stencil-route-title>
    <tf-navigation size="small"></tf-navigation>
    <div class="component-list__wrapper">
      <table></table>
    </div>
    <tf-form-integration></tf-form-integration>
  </mock:shadow-root>
</tf-component-list>
`;

exports[`ComponentListComponent renders with values 1`] = `
<tf-component-list>
  <mock:shadow-root>
    <stencil-route-title pagetitle="Themeful - Components"></stencil-route-title>
    <tf-navigation size="small"></tf-navigation>
    <div class="component-list__wrapper">
      <table></table>
    </div>
    <tf-form-integration></tf-form-integration>
  </mock:shadow-root>
</tf-component-list>
`;
